### PR TITLE
VideoPress: Add endpoint for returning today's video stats

### PR DESCRIPTION
**Pull request by lhkowalski**
Relates to #26479.

#### Changes proposed in this Pull Request:

* Register `videopress/v1/stats` endpoint to return general video stats
* Return the counter of plays for today, using the information provided by the WPCOM REST API
* The plan was to add a counter for all-time plays, but we still need to sort out how to request it, so it's not included here

Pending issues:

* which capability the user needs to have to see this data?
* how to make this work inside WPCOM / make sure the class got loaded (similar to #26406)

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

N/A.

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Use a site with VideoPress installed
* Do a `GET` request to `yoursite.com/wp-json/videopress/v1/stats`, for example:

```
curl -X GET -u "user:pass" http://yoursite.com/wp-json/videopress/v1/stats | jq
```

* The expected response is similar to this:

```
{
    "plays": {
        "today": 1
    }
}
```

* It's useful to have some media available on the site with usage information, so make sure you uploaded at least one video, put it on a post and do at least one play on it

### CODE REVIEW

File: projects/packages/videopress/src/class-videopress-rest-api-v1-stats.php [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+		);
+	}
+
+	/**
+	 * Checks wether the user have permissions to see stats
+	 *
+	 * @return boolean
+	 */
+	public static function permissions_callback() {
+		return current_user_can( 'read' ); // TODO: confirm this
```
Comment (retrofox) [inline projects/packages/videopress/src/class-videopress-rest-api-v1-stats.php:48]: cc @fgiannar @leogermani, you folk can help with this?

Comment (retrofox) [reply projects/packages/videopress/src/class-videopress-rest-api-v1-stats.php:48]: image

This is nice

**Review by retrofox — DISMISSED** [review]
LGTM. Please, let's keep an open eye on security issues, but I think it's ok.

By the way, we'd probably like to extract a method to get the stats to get it in the backend size without the need to hit the endpoint from the client.

**Comment by lhkowalski** [comment]
> we'd probably like to extract a method to get the stats to get it in the backend size without the need to hit the endpoint from the client

@retrofox I refactored the code to move the stats requesting to a new `Stats` class. Let me know what you think :+1:

**Review by retrofox — APPROVED** [review]
LGTM. Thanks for addressing the request about extracting the method.

**Pull request merged**